### PR TITLE
feat(inputs.sql): Add oracle driver

### PR DESCRIPTION
**Pull request by michelle-costa**
# Required for all PRs

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

Added the Oracle drive to the SQL plugin to monitor this type of database

### CODE REVIEW

**Comment by srebhan** [comment]
Thanks for the PR @michelle-costa! Please sign the CLA so we can review this PR!

**Comment by powersj** [comment]
!signed-cla

Looks like you did sign the CLA, but instead of your username you put a link to your profile.

**Comment by powersj** [comment]
!signed-cla

**Comment by powersj** [comment]
@michelle-costa if you could get the remaining tests to pass that would be the test next.

Looks like the current failure is a not clean go.mod, so please run:

```
go mod tidy
```

and check in any changes.

Thanks!

File: go.mod [hunk 1]
```diff
@@ -213,6 +213,8 @@ require (
 	modernc.org/sqlite v1.24.0
 )
 
+require github.com/sijms/go-ora/v2 v2.7.18 // indirect
+
 require (
```
Comment (srebhan) [inline go.mod:218]: Please fuse the two sections.

File: plugins/inputs/sql/sql.go [hunk 2]
```diff
@@ -375,6 +377,13 @@ func (s *SQL) setupConnection() error {
 	dsn := string(dsnSecret)
 	config.ReleaseSecret(dsnSecret)
 
+	// Replaces special characters in the oracle connection string
+	if s.driverName == "oracle" {
+		escaped := url.PathEscape(dsn)
+		dsn = strings.ReplaceAll(escaped, "%2F", "/")
+		s.Log.Debugf("Connection %#v", dsn)
```
Comment (srebhan) [inline plugins/inputs/sql/sql.go:384]: Please remove this debugging statement as it has the potential to leak sensitive information encoded in the DSN like credentials...

**Review by srebhan — COMMENTED** [review]
Thanks for adding the Oracle driver @michelle-costa! I do have two small comments and a request: Can you please also document the new driver in [docs/SQL_DRIVERS_INPUT.md](https://github.com/influxdata/telegraf/blob/master/docs/SQL_DRIVERS_INPUT.md)!? Furthermore, please run `make tidy` before checking in the code to keep our CI happy...

File: go.mod (hunk 1 above)
Comment (michelle-costa) [reply go.mod]: Changes made

File: plugins/inputs/sql/sql.go (hunk 2 above)
Comment (michelle-costa) [reply plugins/inputs/sql/sql.go]: Changes made

**Comment by srebhan** [comment]
@michelle-costa please run `make tidy` and check-in the changes and we are good to go...

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Thanks for you contribution @michelle-costa!

**Pull request merged**